### ISSUE TITLE
Reading XLS with password protection

### ISSUE THREAD

**Issue by DeyV** [PHPOffice/PhpSpreadsheet#448]
This is:

```
- [ ] a bug report
- [x] a feature request
 
```

### What is the expected behavior?

XLS with file password protection should be readable. 

### What is the current behavior?

Only default password ( VelvetSweatshop ) is used

### What are the steps to reproduce?

Please provide a [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

	$filename = './data/Reader/XLS/with_password.xls';
	$reader = new \PhpOffice\PhpSpreadsheet\Reader\Xls();
	$reader->setWorkbookPassword('secret');  // or other method for set password 

	$sheet = $reader->load($filename);
         // sheet should be loaded   

```

### Which versions of PhpSpreadsheet and PHP are affected?

1.2.0 - master

**Comment by pcoughlin** [comment]
Has this been included in the current version?
This would solve a problem I am having.
Thanks.

**Comment by mrdanneedham** [comment]
+1 for workbook password, this would really help with GDPR. Been doing some research, perhaps password protected .ods would be easier as this is more documented than excel approach to password protection?

**Comment by ghsolutions** [comment]
+1 for workbook protection with password for gdpr

**Comment by nelero** [comment]
+1, no news on this ?

**Comment by makbul-khan** [comment]
still waiting for this to get fixed.

**Comment by fanzhaogui** [comment]
+1 mark. zip can do this,  unexpected

**Comment by isubhamjha** [comment]
Can anyone please help me, I'm trying to unlock a password protected file with PHP. Can't find any library

**Comment by ngocquang** [comment]
You can try **msoffcrypto-tool** is a Python tool and library for decrypting encrypted MS Office files with a password. And use PhpSpreadsheet to read the unencrypted file in a temporary folder.
I used it in some projects, work great.

**Comment by richdear** [comment]
It would be great if PhpSpreadsheet has functionality to read password protected file with password specified..

**Comment by jaydadhania08** [comment]
I welcome you to check out my [PHPDecryptXLSXWithPassword](https://github.com/jaydadhania08/PHPDecryptXLSXWithPassword) repo. It can decrypt an XLSX file with given password and generate an unprotected (open) XLSX file. Then you can use it with PHPSpreadsheet.

**Comment by Get-Seb** [comment]
> I welcome you to check out my [PHPDecryptXLSXWithPassword](https://github.com/jaydadhania08/PHPDecryptXLSXWithPassword) repo. It can decrypt an XLSX file with given password and generate an unprotected (open) XLSX file. Then you can use it with PHPSpreadsheet. Edit: it also works on DOCX and PPTX files.

HI, awesome work. 
May I know, is it able to read content inside the xls file instead of creating a new unprotected file because i feel it kind of defeats the purpose of having a password protected file
If possible mind sharing the flow :)
Thank you

**Comment by jaydadhania08** [comment]
@Get-Seb I agree, but this is just a script to decrypt the file, and not actually read/process it like PHPOffice does. You could delete the decrypted file (`unlink()`) at the end of your script/function. Alternatively, the decrypted binary content is stored in [`$decrypted`](https://github.com/jaydadhania08/PHPDecryptXLSXWithPassword/blob/5d841ad0a33e621ba67a7cff13fc94cd5aa740c3/PHPDecryptXLSXWithPassword.php#L109), you could skip writing it to a file and somehow (by creating a file stream maybe? idk) feed it directly to PHPOffice.
